mobile/src/main/java/ch/epfl/sdp/mobile/ui/puzzles/PuzzleSelectionScreen.kt

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.sdp.mobile.ui.puzzles
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.state.PuzzleInfoAdapter
+
+/**
+ * Composable that composes the PlayScreen expanded to include history of matches
+ * @param state the [PuzzleSelectionScreenState] to manage composable content
+ * @param modifier the [Modifier] for this composable.
+ * @param key the key to uniquely identify a [PuzzleInfo] list item
+ * @param contentPadding The [PaddingValues] to apply to the content
+ */
+@Composable
+fun PuzzleSelectionScreen(
+    state: PuzzleSelectionScreenState<PuzzleInfoAdapter>,
+    modifier: Modifier = Modifier,
+    key: ((PuzzleInfo) -> Any)? = null,
+    contentPadding: PaddingValues = PaddingValues()
+) {
+
+  val strings = LocalLocalizedStrings.current
+  val lazyListState = rememberLazyListState()
+  val targetShadow =
+      if (lazyListState.firstVisibleItemIndex >= 1 ||
+          lazyListState.firstVisibleItemScrollOffset > 0)
+          4.dp
+      else 0.dp
+  val shadow by animateDpAsState(targetShadow)
+
+  Scaffold(
+      topBar = {
+        Text(
+            text = strings.puzzleUnsolvedPuzzles,
+            color = MaterialTheme.colors.primary,
+            style = MaterialTheme.typography.h4,
+            modifier =
+                Modifier.shadow(shadow)
+                    .background(MaterialTheme.colors.background)
+                    .fillMaxWidth()
+                    .padding(top = 32.dp, start = 16.dp, end = 16.dp, bottom = 8.dp),
+        )
+      },
+      modifier = modifier.padding(contentPadding),
+  ) { innerPadding ->
+    LazyColumn(
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        state = lazyListState,
+        contentPadding = innerPadding,
+        modifier =
+            Modifier.semantics { contentDescription = strings.puzzleListContentDescription }) {
+      items(state.puzzles, key) { puzzle ->
+        PuzzleListInfo(puzzle, { state.onPuzzleItemClick(puzzle) })
```

Comment:
(nitpick, just for readability)

```suggestion
        PuzzleListInfo(puzzle, onClick = { state.onPuzzleItemClick(puzzle) })
```

Reply:
https://github.com/epfl-SDP/android/pull/300/commits/d58b4c103883b86b45e1fad975d3d20853b8373d